tests/test_init.py

```diff
@@ -17,62 +16,93 @@
 )
 from rpdk.core.project import Project
 
+from .utils import add_dummy_language_plugin, dummy_parser, get_args, get_mock_project
+
 PROMPT = "MECVGD"
 ERROR = "TUJFEL"
 
 
-def test_init_method_interactive_language():
+def test_init_method_interactive():
     type_name = object()
     language = object()
 
-    args = Mock(spec_set=["force", "language"])
-    args.force = False
-    args.language = None
-
-    mock_project = Mock(spec=Project)
-    mock_project.load_settings.side_effect = FileNotFoundError
-    mock_project.settings_path = ""
-    mock_project.root = Path(".")
+    args = get_args(interactive=True)
+    mock_project, patch_project = get_mock_project()
 
-    patch_project = patch("rpdk.core.init.Project", return_value=mock_project)
     patch_tn = patch("rpdk.core.init.input_typename", return_value=type_name)
     patch_l = patch("rpdk.core.init.input_language", return_value=language)
 
     with patch_project, patch_tn as mock_tn, patch_l as mock_l:
-        init(args)
+        main(args_in=["init"])
 
     mock_tn.assert_called_once_with()
     mock_l.assert_called_once_with()
 
     mock_project.load_settings.assert_called_once_with()
-    mock_project.init.assert_called_once_with(type_name, language)
+    mock_project.init.assert_called_once_with(
+        type_name,
+        language,
+        args.settings,
+    )
     mock_project.generate.assert_called_once_with()
 
 
-def test_init_method_noninteractive_language():
-    type_name = object()
+def test_init_method_noninteractive():
+    add_dummy_language_plugin()
+
+    args = get_args()
+    mock_project, patch_project = get_mock_project()
+
+    patch_tn = patch("rpdk.core.init.input_typename")
+    patch_l = patch("rpdk.core.init.input_language")
+    patch_get_parser = patch(
+        "rpdk.core.init.get_parsers", return_value={"dummy": dummy_parser}
+    )
+
+    # pylint: disable=C0301
+    with patch_project, patch_tn as mock_tn, patch_l as mock_l, patch_get_parser as mock_parser:  # noqa: B950
+        main(args_in=["init", "--type-name", args.type_name, args.language, "--dummy"])
+
+    mock_tn.assert_not_called()
+    mock_l.assert_not_called()
+    mock_parser.assert_called_once()
+
+    mock_project.load_settings.assert_called_once_with()
+    mock_project.init.assert_called_once_with(
+        args.type_name,
+        args.language,
+        args.settings,
```

Comment:
usually args doesn't have a settings member, correct? let's keep args as simple as possible and in line with what the cli expects its structure to be. it looks like here it is only used to keep track of test values. can always pull that logic out of the get_args method and inline it here.